ambry-store/src/test/java/com.github.ambry.store/IndexTest.java

```diff
@@ -2222,31 +2319,70 @@ private void indexSegmentRollOverTest(boolean rollOverWithPutRecord) throws Stor
     assertEquals("Index roll over should have happened ", indexCount + 1, state.index.getIndexSegments().size());
     currentEndOffset = state.index.getCurrentEndOffset();
     indexEntries.addAll(
-        addPutEntries(currentEndOffset, null, 2, CuratedLogIndexState.PUT_RECORD_SIZE, Utils.Infinite_Time));
+        addPutEntries(currentEndOffset, null, 2, CuratedLogIndexState.PUT_RECORD_SIZE, Utils.Infinite_Time, 10));
     assertEquals("Index roll over should not have happened ", indexCount + 1, state.index.getIndexSegments().size());
     // verify index values
     verifyIndexValues(indexEntries);
   }
 
   /**
+   * Adds entries to the index and asserts that the number of index segments and the persistedEntrySizes of the latest
+   * segment are as expected.
+   * @param indexEntries the list of {@link IndexEntry} to fill with the entries that are added to the index
```

Comment:
Just passing around the same array that gets filled by multiple invocations of this method. It is referred to in the end for verification and I think it is simpler this way.